main.py

```diff
@@ -26,36 +29,265 @@ def info() -> Dict[str, List[str]]:
+    form = ClassificationForm(request)
+    await form.load_data()
+    image_id = form.image_id
+
+    # Generate the histogram
+    histogram_base64 = generate_histogram(image_id)
+
+    return templates.TemplateResponse(
+        "histogram_output.html",
+        {
+            "request": request,
+            "image_id": image_id,
+            # Pass the histogram base64 to the context
+            "histogram_base64": histogram_base64,
         },
     )
+
+
+@app.get("/transformations")
+def create_transformation(request: Request):
+    return templates.TemplateResponse(
+        "transformation_select.html",
+        {"request": request, "images": list_images(), "models": Configuration.models},
+    )
+
+
+# This function is used to delete the temporary image created after the transformation
+def delete_temp_img(temp_path):
+    time.sleep(1)
+    os.remove("app/static/imagenet_subset/" + temp_path)
+
+
+# This function is used to apply the transformation to the image and classify it
+@app.post("/transformations")
+async def request_transformation(request: Request, background_tasks: BackgroundTasks):
+    form = TransformationForm(request)
+    await form.load_data()
+    image_id = form.image_id
+    model_id = form.model_id
+
+    # Open the image and apply the transformations
+    with Image.open("app/static/imagenet_subset/" + image_id) as img:
+        img = ImageEnhance.Color(img).enhance(form.color)
+        img = ImageEnhance.Brightness(img).enhance(form.brightness)
+        img = ImageEnhance.Contrast(img).enhance(form.contrast)
+        img = ImageEnhance.Sharpness(img).enhance(form.sharpness)
+
+        # Save the transformed image
+        temp_path = f"temp.{image_id}"
+        img.save("app/static/imagenet_subset/" + temp_path)
+        img.close()
+
+        classification_scores = classify_image(
+            model_id=model_id, img_id=temp_path)
+
+        # Save the classification scores to a JSON file
+        with open("app/static/transformationScores.json", "w") as json_file:
+            json.dump(classification_scores, json_file)
+            json_file.close()
+
+        # Render the response
+        response = templates.TemplateResponse(
+            "transformation_output.html",
+            {
+                "request": request,
+                "image_id": temp_path,
+                "classification_scores": json.dumps(classification_scores),
+            },
+        )
+
+        # Delete the temporary image after the response is sent
+        background_tasks.add_task(delete_temp_img, temp_path)
+        return response
+
+
+@app.post("/upload/")
+async def create_upload_file(
+    request: Request,
+    image_id: UploadFile = File(...)
+):
+    if (
+        image_id.filename.endswith(".jpg") |
+        image_id.filename.endswith(".JPEG") |
+        image_id.filename.endswith(".jpeg") |
+        image_id.filename.endswith(".png") |
+        image_id.filename.endswith(".webp")
+    ):
```

Comment:
Suggestion (you don't have to change it): it's not safe to check the file format by only looking at its extension. A safer method is to check the magic number, for instance with [python-magic](https://github.com/ahupp/python-magic).